### DISCUSSION TITLE
Hide the Publish Date for some statuses

### DISCUSSION THREAD

**Discussion by stevejburge** [publishpress/PublishPress-Planner#1309] [Feature Requests]
This looks like a good idea.

We probably couldn't remove the date field, but we could hide it:

> I created new statuses that are essentially various draft statuses in our workflow, but as soon as I put a post in any of them it automatically publishes or forces a publish time and date.
> 
> I'd like to be able to just have various draft statuses that move through our writing process without them having to be live or a forced time to go live in case it's not done.

**Comment by mgurtzweiler** [comment]
Any update on this issue? We're running into this using Custom Statuses and posts getting published in the past.

We have a "Media Required" draft state and as soon as a writer sets this status the PublishDate is set.

**Comment by stevejburge** [comment]
@mgurtzweiler You can hide the status with the Editor Features option in PublishPress Capabilities. I'll close this issue for that reason

**Comment by mgurtzweiler** [comment]
What would hiding the status accomplish? We use the status fields to move posts through the workflow. The issue is setting the publish date when a status is assigned.

**Comment by mgurtzweiler** [comment]
https://developer.wordpress.org/reference/functions/register_post_status/

It looks like setting the date_floating option to true would solve this.

**Comment by andergmartins** [comment]
Thanks @mgurtzweiler, this seems to be a Gutenberg issue, that is not respecting the date_floating param. We are already using date_floating = true: 

https://github.com/publishpress/PublishPress/blob/master/modules/custom-status/custom-status.php#L328

If using the classic editor, the issue do not happens. But if using Gutenberg it sets the publishing date by itself if we change the post status for any custom status.

We are still looking for a fix.

**Comment by andergmartins** [comment]
We probably need to fix this submitting a PR to Gutenberg fixing the function `isEditedPostDateFloating`:

https://github.com/WordPress/gutenberg/blob/26c8b7149091a315a0da09544cc74cfdc5fbd9c3/packages/editor/src/store/selectors.js#L666-L683

That function is checking date floating for a hardcoded list of statuses. It is not recognizing custom statuses. 

There is an issue for this already: https://github.com/WordPress/gutenberg/issues/26984
